src/main/resources/assets/js/components/AppListComponent.jsx

```diff
@@ -29,23 +33,36 @@ var AppListComponent = React.createClass({
     collection.sort();
   },
 
+  filterCollection: function (filter) {
+    this.setState({filter:filter||''});
```

Comment:
I wouldn't handle the fallback to "" at this point, would check it where it is needed.